backend/src/math/math.ts

```diff
@@ -0,0 +1,23 @@
+export function dotProduct(a: number[], b: number[]) {
```

Comment:
nit: consider also annotating the return value like
```suggestion
export function dotProduct(a: number[], b: number[]): number {
```